### DISCUSSION TITLE
useContractRead always runs on page load

### DISCUSSION THREAD

**Discussion by 0xconstant** [wevm/wagmi#831] [Q&A]
I'm a bit confused about how the enable property works on the useContractRead hook. 

My desired functionality is to only run the hook on a button click. I use the refetch method on the hook in an onClick handler and have set the enabled property to false. However, the hook still runs on every page load. Is there an example that shows how useContractRead can be used only on button clicks? Thanks.

```
const contractRead = useContractRead({
    addressOrName: myAddress,
    contractInterface: myAbi,
    functionName: "myFunc",
    args: [props.arg1, props.arg2],
    enabled: false,
  });

  const handleClick = async () => {
    await contractRead.refetch();
  };
```

Any guidance is appreciated. Thanks.

**Comment by dev-cz** [comment]
I dont believe there is a way to handle contractreads on click. Been trying to do this myself and from other posts it doesnt seem possible.

I have it working somewhat but in a very dirty way, but not 100% as sometimes it works, other times it doesn't or it renders fewer/more hooks than before. So yeah.

**Comment by L-TheG** [comment]
Have you tried the "enabled: false" flag in the useContractRead config?
```
const {refetch: readPubPrice, data} = useContractRead({
    addressOrName: yourContractAddress,
    contractInterface: yourAbi,
    functionName: 'yourFnName',
    enabled: false,
});
```

**Reply by rsodre** [reply]
I also use enabled to trigger contract reads.

**Comment by meri-maki** [comment]
Hi! Is there currently a solution to use contractRead "on demand"? like write() in useContractWrite

**Comment by hellowodl** [comment]
If I wouldn't like the data to be shown right now, I'd display it in a subcomponent that gets shown conditionally.
The same can be done if there's arguments involved.

If I'd like the data to be directly available in this component as it directly modifies the data shown in this component, I'd use @wagmi/core

@wagmi/react is just a @tanstack/query hooks layer written on top of @wagmi/core, so you can freely use that without drawbacks.
